### PR TITLE
Add SC V0 tests for relaxed restrictions

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Add tests for V0 contracts that test https://github.com/Concordium/concordium-node/pull/573.

## Changes

- Add test of relaxed restrictions for V0 contracts.
- Move existing v0 contract tests to v0 folder.
  - I didn't think the new test fit well into the existing file with v0 tests.
  - So I created a new file (by modifying the v1 one) and moved the existing v0 tests.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by abizjak — APPROVED** [review]

File: concordium-consensus/testdata/contracts/relaxed-restrictions.wat [hunk 1]
```diff
@@ -0,0 +1,120 @@
+;; Test for the relaxation of restrictions on the parameter size, return value size, and number of logs.
```
Comment (EmilLa1) [inline concordium-consensus/testdata/contracts/relaxed-restrictions.wat:1]:
```suggestion
;; Testing the relaxed restrictions wrt. parameter size, return value size, and number of logs introduced with protocol version 5.
```

File: concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs [hunk 2]
```diff
@@ -0,0 +1,267 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeApplications #-}
+{-| This module tests the relaxed smart contract restrictions introduced in P5.
```
Comment (EmilLa1) [inline concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs:4]: Maybe add "... for V0 contracts" somewhere here.

File: concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+ensureSuccess :: TVer.BlockItemWithStatus -> Types.TransactionSummary -> Expectation
+ensureSuccess _ Types.TransactionSummary{..} = checkSuccess "Update failed" tsResult
+
+checkSuccess :: [Char] -> Types.ValidResult -> IO ()
+checkSuccess msg Types.TxReject{..} = assertFailure $ msg ++ show vrRejectReason
+checkSuccess _ _ = return ()
+
+tests :: Spec
+tests = do
+  describe "V1: Relax restrictions. Test in PV4." $
```
Comment (EmilLa1) [inline concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs:264]:
```suggestion
  describe "V0: Relax restrictions. Test in PV4." $
```

File: concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    mkSpecs testCasesPV4
+  describe "V1: Relax restrictions. Test in PV5." $
```
Comment (EmilLa1) [inline concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V0/RelaxedRestrictions.hs:266]:
```suggestion
  describe "V0: Relax restrictions. Test in PV5." $
```

**Review by EmilLa1 — COMMENTED** [review]
Great work!

**Review by EmilLa1 — APPROVED** [review]
Great stuff!

**Pull request merged**